### ISSUE TITLE
the total_cost and wd_cost become nan.

### ISSUE THREAD

**Issue by bobzhang123** [google-research/ssl_detection#9]
hi, when I test your code with train_stg1.sh and compute the teacher model. the logs show that the total_cost and wd_cost become Nan, I did not change any code. 
the data and the gpu is as follows:
DATASET='coco_train2017.1@10'
CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7
![image](https://user-images.githubusercontent.com/13234596/87137760-0456b980-c2d0-11ea-934b-27ef6cfa08ec.png)

**Comment by zizhaozhang** [comment]
There is a rare case a single forward can make the loss be nan, and when it is accumulated to the moving average of total_cost and wd_cost variables, it makes the stdout being nan. We haven't fix this internal issue but it does not affect the training.

**Comment by bobzhang123** [comment]
> There is a rare case a single forward can make the loss be nan, and when it is accumulated to the moving average of total_cost and wd_cost variables, it makes the stdout being nan. We haven't fix this internal issue but it does not affect the training.

Thank you for your reply. In my case, It actually have a bad affect on my result. After the model is trained, when I  run 'eval_stg1.sh' , the results eg mAP, AP50... all become zero.

**Comment by zizhaozhang** [comment]
From your screenshot, the recall results seem correct. So I am wondering if the problem is not because of nan but others. All scripts we provided are tested before. Can you during training, the COCO eval matrics in tensorboard are correct?

If not, could you pls list all the exact commands you run. I can not inspect based on currrent information

**Comment by bobzhang123** [comment]
> From your screenshot, the recall results seem correct. So I am wondering if the problem is not because of nan but others. All scripts we provided are tested before. Can you during training, the COCO eval matrics in tensorboard are correct?
> 
> If not, could you pls list all the exact commands you run. I can not inspect based on currrent information

Hi，
this is my training script in 'train_stg1.sh':
![image](https://user-images.githubusercontent.com/13234596/87617649-802a8900-c74a-11ea-91f3-d177b255fe1d.png)
this is my eval script in 'eval_stg1.sh':
![image](https://user-images.githubusercontent.com/13234596/87617800-d13a7d00-c74a-11ea-9a1a-940ae69ffb97.png)
I do not have much modification, only from ’model-18000‘ to ’model-6000‘：
the evaluation result is as follows:
![image](https://user-images.githubusercontent.com/13234596/87617931-24accb00-c74b-11ea-9dc8-6d1e147837b8.png)
what a strange problem!

**Comment by zizhaozhang** [comment]
Thanks. We haven't met this issue in our training and seems other users also do not report this. I would suggest

1. whether your tensorflow version (1.14) mets our listed requirement
2. whether you follow the right data prepartion. 

No need to tune parameteres to avoid this issue, the default parameters should work.

**Comment by bobzhang123** [comment]
> Thanks. We haven't met this issue in our training and seems other users also do not report this. I would suggest
> 
> 1. whether your tensorflow version (1.14) mets our listed requirement
> 2. whether you follow the right data prepartion.
> 
> No need to tune parameteres to avoid this issue, the default parameters should work.

Hi， 
I have solve the porblem, I use the released branch 'dependabot/pip/tensorflow-gpu-1.15.2', and also my tf version is 1.15.2, I downgrage the tf version to 1.14.0 and the python version to 3.6.8, the NaN problem and the OOM problem is solved!
It seems that the released branch 'dependabot/pip/tensorflow-gpu-1.15.2' is not robust and still have some bugs.
